src/module_ncio.f90

```diff
@@ -826,528 +837,527 @@ subroutine close_dataset(dset,errcode)
-  !          program will stop if a nonzero error code returned
-  !          from netcdf library.
-
-  !subroutine write_vardata(dset,varname,values,nslice,slicedim,errcode)
-  ! write data (in array values) to variable varname in dataset dset.
-  !
-  ! dset:    Input dataset instance returned by open_dataset/create_dataset.
-  ! varname: Input string name of variable.
-  ! values:  Array with variable data.  Must be
-  !          an allocatable array with same rank
-  !          as variable varname (or 1 dimension less).
-  ! nslice:  optional index along dimension slicedim
-  ! slicedim: optional, if nslice is set, index of which dimension to slice with
-  !          nslice, default is ndims
-  ! ncstart: optional, if ncstart and nccount are set, manually specify the
-  !          start and count of netCDF write
-  ! nccount: optional, if ncstart and nccount are set, manually specify the
-  !          start and count of netCDF write
-  ! errcode: optional error return code.  If not specified,
-  !          program will stop if a nonzero error code returned
-  !          from netcdf library.
-
-  !subroutine read_attribute(dset, attname, values, varname, errcode)
-  ! read attribute 'attname' return in 'values'.  If optional
-  ! argument 'varname' is given, a variable attribute is returned.
-  ! if the attribute is a 1d array, values should be an allocatable 1d
-  ! array of the correct type.
-
-  !subroutine write_attribute(dset, attname, values, varname, errcode)
-  ! write attribute 'attname' with data in 'values'.  If optional
-  ! argument 'varname' is given, a variable attribute is written.
-  ! values can be a real(4), real(8), integer, string or 1d array.
-
+  !> subroutine read_vardata(dset,varname,values,nslice,slicedim,errcode)
+  !! read data from variable varname in dataset dset, return in it array values.
+  !!
+  !! dset:    Input dataset instance returned by open_dataset/create_dataset.
+  !! varname: Input string name of variable.
+  !! values:  Array to hold variable data.  Must be
+  !!          an allocatable array with same rank
+  !!          as variable varname (or 1 dimension less).
+  !! nslice:  optional index along dimension slicedim
+  !! slicedim: optional, if nslice is set, index of which dimension to slice with
+  !!          nslice, default is ndims
+  !! ncstart: optional, if ncstart and nccount are set, manually specify the
+  !!          start and count of netCDF read
+  !! nccount: optional, if ncstart and nccount are set, manually specify the
+  !!          start and count of netCDF read
+  !! errcode: optional error return code.  If not specified,
+  !!          program will stop if a nonzero error code returned
+  !!          from netcdf library.
+  !!
+  !! subroutine write_vardata(dset,varname,values,nslice,slicedim,errcode)
+  !! write data (in array values) to variable varname in dataset dset.
+  !!
+  !! dset:    Input dataset instance returned by open_dataset/create_dataset.
```

Comment:
Turn these into proper parameter descriptions by using the @param tag.

```
!! @param[in] dset dset description
!! @param[in] varname name of variable

```etc.

Reply:
changes have been made based on your comments